lib/gcloud/dns/zone/transaction.rb

```diff
@@ -0,0 +1,142 @@
+        #     tx.add     "example.com.", "A",  86400, "1.2.3.4"
+        #     tx.remove  "example.com.", "TXT"
+        #     tx.replace "example.com.", "MX", 86400, ["10 mail1.example.com.",
+        #                                              "20 mail2.example.com."]
+        #   end
+        #
+        def add name, type, ttl, data
+          @additions += Array(@zone.record(name, type, ttl, data))
+        end
+
+        ##
+        # Removes records from the Zone. The records are looked up before they
+        # are removed.
+        #
+        # === Parameters
+        #
+        # +name+::
+        #   The owner of the record. For example: +example.com.+. (+String+)
+        # +type+::
+        #   The identifier of a {supported record
+        #   type}[https://cloud.google.com/dns/what-is-cloud-dns].
+        #   For example: +A+, +AAAA+, +CNAME+, +MX+, or +TXT+. (+String+)
+        #
+        # === Example
+        #
+        #   require "gcloud"
+        #
+        #   gcloud = Gcloud.new
+        #   dns = gcloud.dns
+        #   zone = dns.zone "example-zone"
+        #   zone.update do |tx|
+        #     tx.add     "example.com.", "A",  86400, "1.2.3.4"
+        #     tx.remove  "example.com.", "TXT"
+        #     tx.replace "example.com.", "MX", 86400, ["10 mail1.example.com.",
+        #                                              "20 mail2.example.com."]
+        #   end
+        #
+        def remove name, type
+          @deletions += @zone.records(name: name, type: type).all
+        end
+
+        ##
+        # Replaces existing records on the Zone. Records matching the +name+ and
+        # +type+ are replaced.
+        #
+        # === Parameters
+        #
+        # +name+::
+        #   The owner of the record. For example: +example.com.+. (+String+)
+        # +type+::
+        #   The identifier of a {supported record
+        #   type}[https://cloud.google.com/dns/what-is-cloud-dns].
+        #   For example: +A+, +AAAA+, +CNAME+, +MX+, or +TXT+. (+String+)
+        # +ttl+::
+        #   The number of seconds that the record can be cached by resolvers.
+        #   (+Integer+)
+        # +data+::
+        #   The resource record data, as determined by +type+ and defined in RFC
+        #   1035 (section 5) and RFC 1034 (section 3.6.1). For example:
+        #   +192.0.2.1+ or +example.com.+. (+String+ or +Array+ of +String+)
+        #
+        # === Example
+        #
+        #   require "gcloud"
+        #
+        #   gcloud = Gcloud.new
+        #   dns = gcloud.dns
+        #   zone = dns.zone "example-zone"
+        #   zone.update do |tx|
+        #     tx.add     "example.com.", "A",  86400, "1.2.3.4"
+        #     tx.remove  "example.com.", "TXT"
+        #     tx.replace "example.com.", "MX", 86400, ["10 mail1.example.com.",
+        #                                              "20 mail2.example.com."]
+        #   end
+        #
+        def replace name, type, ttl, data
```

Comment:
What about passing Record instance(s) as well as passing just the property values? For me, the preferred way to change an existing record would be to retrieve it, modify the value I want to change, and then pass it to this method.

```ruby
mx_record = zone.records(name: "example.com.", type: "MX").all.first
mx_record.ttl = 3600 # change the TTL
zone.update do |tx|
  # do some other adds, replaces, and removes...
   tx.replace mx_record
 end
``` 

Using the Record instance is a convenient way to "pass back" the values I don't want to change. And it makes sense that I would have the instance, otherwise how could I know what the current values are?